src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final BulldozerProducer bulldozerProducer = new BulldozerProducer();
```

Comment:
```suggestion
    private final MachineProducer<BulldozerProducer> bulldozerProducer = new BulldozerProducer();
```
same for others. We should depends on interfaces.